src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -456,6 +502,28 @@ void ImageReplayer<I>::handle_init_remote_journaler(int r) {
     return;
   }
 
+  {
+    Mutex::Locker locker(m_lock);
+
+    m_local_image_ctx->journal->add_listener(
+                                    librbd::journal::ListenerType::RESYNC,
+                                    m_resync_listener);
+
+    bool do_resync = false;
+    r = m_local_image_ctx->journal->check_resync_requested(&do_resync);
+    if (r < 0) {
+      derr << "failed to check if a resync was requested" << dendl;
+    }
+
+    if (do_resync) {
+      Context *on_finish = m_on_start_finish;
+      FunctionContext *ctx = new FunctionContext([this, on_finish](int r) {
+          resync_image(on_finish);
```

Comment:
hmm... I think that a resync should be processed even if the image-replayer start failed. Maybe we are requesting a resync exactly because the image-replayer fails to start...